### ISSUE TITLE
No man page for ed, the standard text editor

### ISSUE THREAD

**Issue by LemmingAvalanche** [NixOS/nixpkgs#32150]
## Issue description

There seems to be no man page for ed, the standard text editor.

### Steps to reproduce

```
$ nix-shell --pure -p ed -p man -p less \
    -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/master.tar.gz
$ man ed
No manual entry for ed
```

## Technical details

```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.58, NixOS, 17.09.1880.ac2bb5684c (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.15`
 - channels(root): `"nixos-17.09.1880.ac2bb5684c"`
 - channels(kristoffer): `"nixpkgs-17.09pre112229.e619ace733"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
```

**Comment by matthewbauer** [comment]
It looks like an issue in how ed installs things:

```
$ tree /nix/store/ann881y0l5f72fim46mnzc13yy4xf6p2-ed-1.14.2/
/nix/store/ann881y0l5f72fim46mnzc13yy4xf6p2-ed-1.14.2/
├── bin
│   ├── ed
│   └── red
└── nix
    └── store
        └── ann881y0l5f72fim46mnzc13yy4xf6p2-ed-1.14.2
            └── share
                ├── info
                │   └── ed.info
                └── man
                    └── man1
                        ├── ed.1
                        └── red.1 -> ed.1

8 directories, 5 files
```

**Comment by berce** [comment]
`pkgs/applications/editors/ed/default.nix` contains:
```
     /* FIXME: Tests currently fail on Darwin:

       building test scripts for ed-1.5...
       testing ed-1.5...
       *** Output e1.o of script e1.ed is incorrect ***
       *** Output r3.o of script r3.ed is incorrect ***
       make: *** [check] Error 127

    */
```
I suspect the next code results in double application of the nix store directory:

```
  installFlags = [ "DESTDIR=$(out)" ];

  configureFlags = [
    "--exec-prefix=${stdenv.cc.prefix}"
    "CC=${stdenv.cc.prefix}cc"
  ];
```

I 'll investigate further.

**Comment by berce** [comment]
This might be related to a comment in `pkgs/applications/editors/ed/default.nix`:
```
     /* FIXME: Tests currently fail on Darwin:

       building test scripts for ed-1.5...
       testing ed-1.5...
       *** Output e1.o of script e1.ed is incorrect ***
       *** Output r3.o of script r3.ed is incorrect ***
       make: *** [check] Error 127

    */
```
The target directory is set with:

```
  installFlags = [ "DESTDIR=$(out)" ];

  configureFlags = [
    "--exec-prefix=${stdenv.cc.prefix}"
    "CC=${stdenv.cc.prefix}cc"
  ];
```
There's another configure flag, `--prefix`, that defaults to `/usr/local`. I think nix automatically replaces that by `$(out)`. Resulting in `$(out)/$(out)`.
`--exec-prefix` would default to '$(prefix)`, but since it was set, the bin files ended up in the right place.

Not setting any installFlags results in a correct tree. It basically lets nix figure it out.
```
  # installFlags = [ "DESTDIR=$(out)" ];

  configureFlags = [
    "--exec-prefix=${stdenv.cc.prefix}"
    "CC=${stdenv.cc.prefix}cc"
  ];
```
I noticed @Ericson2314 very recently replaced `stdenv.cc.prefix` by `stdenv.cc.targetPrefix'.

Todo:
- [ ] Confirm it's ok to not set the `DESTDIR`
- [ ] Check if the `FIXME` comment in the source about Darwin is solved.

**Comment by dezgeg** [comment]
Yes, removing `DESTDIR` is usually the correct option.

**Comment by orivej** [comment]
Fixed by #32322.
